Source/Lib/Codec/EbEncHandle.c

```diff
@@ -240,7 +240,7 @@ static int32_t can_use_intel_avx512()
 // Using bit-fields, the fastest function will always be selected based on the available functions in the function arrays
 uint32_t get_cpu_asm_type()
 {
-    uint32_t asm_type = 0;
+    uint32_t asm_type = 7; //This fixes the case where AVX2 is not getting invoked even when AVX2 is supported in the platform.
```

Comment:
Hi @techvintage ,

Please consider to implemente the checking for AMD CPUs, like what CanUseIntelAVX512() and Check4thGenIntelCoreFeatures() do for Intel CPUs. Your change occationally can work on your platform which supports AVX2, but cannot for other (old) platforms which can only support SSEx or no intrinsics.
In theory, GetCpuAsmType() should check all the x86 CPUs with SSE and/or AVX intrinsics supported, including Intel, AMD (and VIA).

Thanks!